src/UI/Shared/Grid/SonarrPageableCollection.js

```diff
@@ -0,0 +1,42 @@
+var _ = require('underscore');
+var PageableCollection = require('backbone.pageable');
+
+var SonarrPageableCollection = PageableCollection.extend({
+    _makeCollectionEventHandler : function (pageCollection, fullCollection) {
+        this.pageCollection = pageCollection;
+        this.fullCollection = fullCollection;
+        var eventHandler = PageableCollection.prototype._makeCollectionEventHandler.apply(this, arguments);
+
+        return _.wrap(eventHandler, _.bind(this._eventHandler, this));
+    },
+
+    _eventHandler : function (originalEventHandler, event, model, collection, options) {
+        if (event === 'reset') {
+            var currentPage = this.state.currentPage;
+            var pageSize = this.state.pageSize;
+            var pageStart = currentPage * pageSize;
+            var pageEnd = pageStart + pageSize;
+
+            originalEventHandler.apply(this, [].slice.call(arguments, 1));
+
+            var totalPages = Math.ceil(this.state.totalRecords / pageSize);
+
+            if (currentPage > totalPages) {
```

Comment:
why reset to 0? why not snap to last page?

And isn't it possible to simplify this whole piece of code to:
```
var newPage = Math.min(currentPage, totalPages - 1);
if (newPage !== this.state.currentPage) {
   this.getPage(newPage);
}
```